compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/service/validator/InterceptorValidator.java

```diff
@@ -0,0 +1,87 @@
+package io.ballerina.stdlib.graphql.compiler.service.validator;
+
+import io.ballerina.compiler.api.symbols.MethodSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.SymbolKind;
+import io.ballerina.compiler.syntax.tree.ClassDefinitionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.TypeReferenceNode;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.stdlib.graphql.compiler.service.errors.CompilationError;
+import io.ballerina.tools.diagnostics.Location;
+
+import static io.ballerina.stdlib.graphql.compiler.Utils.isRemoteMethod;
+import static io.ballerina.stdlib.graphql.compiler.service.validator.ValidatorUtils.GRAPHQL_INTERCEPTOR;
+import static io.ballerina.stdlib.graphql.compiler.service.validator.ValidatorUtils.INTERCEPTOR_EXECUTE;
+import static io.ballerina.stdlib.graphql.compiler.service.validator.ValidatorUtils.updateContext;
+
+/**
+ * Validate Interceptors in Ballerina GraphQL services.
+ */
+public class InterceptorValidator {
+
+    private final ClassDefinitionNode classDefinitionNode;
+    private final SyntaxNodeAnalysisContext context;
+    private boolean errorOccurred;
+
+    public InterceptorValidator(SyntaxNodeAnalysisContext context, ClassDefinitionNode classDefinitionNode) {
+        this.context = context;
+        this.classDefinitionNode = classDefinitionNode;
+        this.errorOccurred = false;
+    }
+
+    public void validate() {
+        NodeList<Node> members = this.classDefinitionNode.members();
+        for (Node member : members) {
+            if (member.kind() == SyntaxKind.TYPE_REFERENCE) {
+                String typeReference = ((TypeReferenceNode) member).typeName().toString();
+                if (typeReference.equals(GRAPHQL_INTERCEPTOR)) {
+                    validateInterceptorService(members);
+                    break;
+                }
```

Comment:
This is not accurate to check the type name. Check [`isFileUploadParameter()`](https://github.com/ballerina-platform/module-ballerina-graphql/blob/3a7dbfb6ee057d95f91f5b9dbf386f7e32dffbf5/compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/Utils.java#L204) method for example.